src/librustc_ty/needs_drop.rs

```diff
@@ -99,6 +99,23 @@ where
                         }
                     }
 
+                    ty::Generator(_, substs, _) => {
```

Comment:
Do we have assertions somewhere that could catch bugs in `needs_drop` (i.e., catch cases where a type has drop glue but `needs_drop` says no)?

Reply:
Not that I know of, but we have several mir-opt tests for generators, as well as diagnostics tests, which were (correctly) affected by this change.